src/odemis/driver/tescan.py

```diff
@@ -421,6 +421,28 @@ def _single_acquisition(self, channel):
             dt = self._scanner.dwellTime.value * 1e9
             # with self._acquisition_init_lock:
             logging.debug("Acquiring SEM image of %s with dwell time %f ns", res, dt)
+
+            # make sure socket settings are always set
+            self._device.connection.socket_c.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+            self._device.connection.socket_d.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+            # Check if spot mode is required
+            if res == (1, 1):
+                if ((self._scaled_shape != scaled_shape) or
+                        (self._roi != (l, t, r, b)) or
+                        (self._dt != dt) or
+                        (self.pre_res != res)):
+                    self._device.ScStopScan()
+                    # flush remaining data in data buffer
+                    self.flush()
+                    # need to reset
+                    self._device.connection.socket_c.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+                    self._device.connection.socket_d.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+            else:
+                if self.pre_res == (1, 1) or True:  # debug
```

Comment:
We cannot leave this `or True` as is. Either remove it,  or remove the whole if, and the inside block should always be executed. Hopefully we can leave the if. Can you check?

Reply:
It doesn't work if I just delete "or True". So I will remove the whole if.